reactor-netty-http/src/test/java/reactor/netty/http/HttpMetricsHandlerTests.java

```diff
@@ -284,24 +298,27 @@ void testRecordingFailsClientSide(HttpProtocol[] serverProtocols, HttpProtocol[]
 	@ParameterizedTest
 	@MethodSource("httpCompatibleProtocols")
 	void testNonExistingEndpoint(HttpProtocol[] serverProtocols, HttpProtocol[] clientProtocols,
-			@Nullable ProtocolSslContextSpec serverCtx, @Nullable ProtocolSslContextSpec clientCtx) throws Exception {
-		// For HTTP11, we expect to observe 2 DISCONNECTS for client, and 2 DISCONNECT for server.
-		// Else, we expect to observe 2 DISCONNECTS for client, and 1 DISCONNECT for server.
-		boolean isHTTP11 = clientProtocols.length == 1 && clientProtocols[0] == HttpProtocol.HTTP11;
-		int expectedDisconnects = isHTTP11 ? 4 : 3;
-
-		CountDownLatch latch = new CountDownLatch(expectedDisconnects);
-		AtomicReference<CountDownLatch> latchRef = new AtomicReference<>(latch);
-		ConnectionObserver observerDisconnect = observeDisconnect(latchRef);
-
+	                             @Nullable ProtocolSslContextSpec serverCtx, @Nullable ProtocolSslContextSpec clientCtx) throws Exception {
+		CountDownLatch responseSent = new CountDownLatch(1); // response fully sent by the server
+		AtomicReference<CountDownLatch> responseSentRef = new AtomicReference<>(responseSent);
+		ResponseSentHandler responseSentHandler = ResponseSentHandler.INSTANCE;
+		CountDownLatch requestReceived = new CountDownLatch(1); // request fully received by the server
+		AtomicReference<CountDownLatch> requestReceivedRef = new AtomicReference<>(requestReceived);
+		RequestReceivedHandler requestReceivedHandler = RequestReceivedHandler.INSTANCE;
+
+		// the serverReceivedHandler is used to detect when the server has received the last client request content
+		// the serverCompletedHandler is used to detect when the server has sent the last response content
 		disposableServer = customizeServerOptions(httpServer, serverCtx, serverProtocols)
-				.childObserve(observerDisconnect)
+				.doOnConnection(cnx -> responseSentHandler.register(responseSentRef, cnx.channel().pipeline()))
+				.doOnConnection(cnx -> requestReceivedHandler.register(requestReceivedRef, cnx.channel().pipeline()))
```

Comment:
Can we do this with one callback? E.g.

```suggestion
				.doOnConnection(cnx -> {
					responseSentHandler.register(responseSentRef, cnx.channel().pipeline());
					requestReceivedHandler.register(requestReceivedRef, cnx.channel().pipeline());
				})
```